sdl-core/src/main/scala/io/smartdatalake/workflow/dataobject/JdbcTableDataObject.scala

```diff
@@ -161,7 +161,8 @@ case class JdbcTableDataObject(override val id: DataObjectId,
       val newDataType = if (SchemaUtil.isSparkCaseSensitive) df.schema.find(_.name == incrementalOutputColumn.get).get.dataType
       else df.schema.find(_.name.equalsIgnoreCase(incrementalOutputColumn.get)).get.dataType
       if (context.phase == ExecutionPhase.Exec) {
-        val newWatermarkValue = df.agg(max(col(incrementalOutputColumn.get))).head.get(0)
+        val newWatermarkValue = Option(df.agg(max(col(incrementalOutputColumn.get))).head.get(0))
```

Comment:
watermark is probably not the right word for this.
Definition form google: Software watermarking involves embedding a unique identifier within a piece of software, to discourage software theft. 
Maybe bookmark would be better?

Reply:
@Geheiner right, but it's used else where in streaming, e.g. Kafka: https://docs.confluent.io/platform/current/clients/confluent-kafka-dotnet/_site/api/Confluent.Kafka.WatermarkOffsets.html
I think it's more clear if i call it "highWatermark" (which translates to Hochwasserstand) and dont simplify it to watermark. WDJT?

Reply:
@zzeekk @Geheiner I would have proposed highWatermark as well. The term is also used internally by RDMSs.

Reply:
ahh they we are talking watermarks in the context of "Wasserpegelstand" and not "Wasserzeichen".
"A high water mark is a point that represents the maximum rise of a body of water over land. Such a mark is often the result of a flood, but high water marks may reflect an all-time high, an annual high or the high point for some other division of time."
Then I agree with the two of you, highWatermark sounds good!